service/src/main/java/bio/terra/cbas/controllers/MethodsApiController.java

```diff
@@ -301,8 +304,16 @@ public ResponseEntity<MethodListResponse> getMethods(
 
     List<MethodDetails> methodDetails;
     if (methodVersionId != null) {
-      methodDetails =
-          List.of(methodVersionToMethodDetails(methodVersionDao.getMethodVersion(methodVersionId)));
+      try {
+        methodDetails =
+            List.of(
+                methodVersionToMethodDetails(methodVersionDao.getMethodVersion(methodVersionId)));
+      } catch (bio.terra.cbas.common.exceptions.MethodProcessingException
+          | java.net.MalformedURLException
+          | java.net.URISyntaxException
+          | bio.terra.dockstore.client.ApiException e) {
+        return ResponseEntity.badRequest().body(new MethodListResponse().error(e.toString()));
+      }
     } else {
       List<Method> methods =
           methodId == null ? methodDao.getMethods() : List.of(methodDao.getMethod(methodId));
```

Comment:
Should the similar construct hash-ed method url also be added here in this else case?

Reply:
it should be getting the constructed url by way of `methodToMethodDetails` [on line 323](https://github.com/DataBiosphere/cbas/pull/296/files/ebc22f5e2d03d6a1dda034e1ce514659975ca3e2#diff-905f97fe9aa55195961e43b0f23f4b48e35fe77006e1536aaecf034fcdf361fcR323)